test/library/packages/UnitTest/UnitTest.chpl

```diff
@@ -698,6 +702,47 @@ module UnitTest {
         throw new owned AssertionError(tmpString);
       }
     }
+
+    /*Specify Max Number of Locales required to run the test*/
+    proc maxLocales(value: int) throws {
+      this.numMaxLocales = value;
+      if this.numMaxLocales < 1 {
+        throw new owned UnexpectedLocales("Max Locales is less than 1");
+      }
+      if this.numMaxLocales < this.numMinLocales {
+        throw new owned UnexpectedLocales("Max Locales is less than Min Locales");
+      }
+      if value < numLocales {
+        throw new owned TestIncorrectNumLocales("Required Locales ="+value);
+      }
+    }
+
+    /*Specify Min Number of Locales required to run the test*/
+    proc minLocales(value: int) throws {
+      this.numMinLocales = value;
+      if this.numMaxLocales < this.numMinLocales {
+        throw new owned UnexpectedLocales("Max Locales is less than Min Locales");
+      }
+      if value > numLocales {
+        throw new owned TestIncorrectNumLocales("Required Locales ="+value);
+      }
+    }
+
+    /*To add how many locales this test requires*/
+    proc addNumLocales(locales: int ...?n) throws {
+      var canRun =  false;
+      for numLocale in locales {
+        dictDomain.add(numLocale);
+        if numLocale == numLocales {
+          canRun = true;
+        }
+      }
+      if !canRun {
+        var localesStr: string = dictDomain: string;
+        var localesErrorStr: string = "Required Locales ="+localesStr:string;
+        throw new owned TestIncorrectNumLocales(localesErrorStr);
```

Comment:
Hmm, what happens if the user calls addNumlocales multiple times and the first set isn't sufficient for the number of locales that are currently specified?  E.g. if `numLocales = 4` and the user calls `addNumLocales(2); addNumLocales(4);`?

Reply:
As discussed in the gitter channel I will throw an error if user try to call the method again.